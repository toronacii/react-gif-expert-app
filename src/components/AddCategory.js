import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [ value, setValue ] = useState('');

  return (
    <form onSubmit={handleSubmit}>
      <input
      type="text"
      value={value}
      onChange={handleChange} />
    </form>
  );

  function handleChange({ target }) {
    setValue(target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    const category = value.trim();
    if (!category) return;

    setCategories(categories => {
      if (!categories.includes(category)) {
        setValue('');
        return [category, ...categories];
      }
      return categories;
    });
  }
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
