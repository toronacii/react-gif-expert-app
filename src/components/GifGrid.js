import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { gifs, loading } = useFetchGifs(category);

  return <>
    <h3 className="animate__animated animate__flash">{category}</h3>
    { loading && <div className="animate__animated animate__flash">Cargando...</div> }
    <div className="card-grid">
      {gifs.map(({ id, ...props }) => (
        <GifGridItem
          key={id}
          {...props} />)
      )}
    </div>
  </>
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
