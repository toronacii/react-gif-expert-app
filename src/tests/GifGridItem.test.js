import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../components/GifGridItem';

describe('Tests on `GifGridItem` component', () => {
  const title = 'test';
  const url = 'https://test.jpg';
  let wrapper = getWrapper();

  beforeEach(() => {
    wrapper = getWrapper();
  });

  test('GifGridItem should be created', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a p with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have a img with url and alt', () => {
    const img = wrapper.find('img');
    const { src, alt } = img.props();
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should have a animate__fadeIn class', () => {
    const div = wrapper.find('div');
    const hasClass = div.hasClass('animate__fadeIn');
    expect(hasClass).toBe(true);
  });

  function getWrapper() {
    return shallow(<GifGridItem url={url} title={title} />);
  }
});