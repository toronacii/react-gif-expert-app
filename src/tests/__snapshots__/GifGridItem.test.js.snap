// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on \`GifGridItem\` component GifGridItem should be created 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test"
    src="https://test.jpg"
  />
  <p>
    test
  </p>
</div>
`;
